src/Product/WCProductAdapter.php

```diff
@@ -260,6 +260,39 @@ protected function map_wc_product_id(): WCProductAdapter {
 		return $this;
 	}
 
+	/**
+	 * Get the slug for the product.
+	 *
+	 * @param int $product_id The product ID.
+	 * @return string The offer ID
+	 * @since x.x.x
+	 * @see https://developers.google.com/shopping-content/reference/rest/v2.1/products#Product.FIELDS-table
+	 */
+	public static function get_offer_id( int $product_id ): string {
+		/** @var ProductHelper $product_helper  */
+		$product_helper = woogle_get_container()->get( ProductHelper::class );
+		$product        = wc_get_product( $product_id );
+		// Check if the product has the google_ids (mc ids) metadata set in the DB. If it has the google_ids we respect the offerID set in the metadata.
+		$has_mc_ids = $product_helper->is_product_synced( $product );
+
+		if ( $has_mc_ids ) {
+			$mc_ids = $product_helper->get_synced_google_product_ids( $product );
+			return $product_helper->get_offer_id_from_mc_id( reset( $mc_ids ) );
```

Comment:
`reset` returns `false` if the array is empty, but `get_offer_id_from_mc_id` forces it's parameter to be a string. Do we need some additional checking to confirm we are getting the expected data?